File: 0002. Add Two Numbers/swift/Solution.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init() { self.val = 0; self.next = nil; }
 *     public init(_ val: Int) { self.val = val; self.next = nil; }
 *     public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
 * }
 */
class Solution {
    func addTwoNumbers(_ l1: ListNode?, _ l2: ListNode?) -> ListNode? {
        var head: ListNode? = nil
        var prev: ListNode? = nil
        var value = 0
        
        var p1 = l1
        var p2 = l2
        while p1 != nil || p2 != nil || value > 0 {
            value += (p1?.val ?? 0) + (p2?.val ?? 0)
            
            var newNode = ListNode(value % 10)
            if prev != nil {
                prev?.next = newNode
            } else {
                head = newNode
            }
            prev = newNode
            
            p1 = p1?.next
            p2 = p2?.next
            value /= 10
        }
        return head
    }
}

File: 0002. Add Two Numbers/c#/Solution.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        ListNode head = null;
        ListNode prev = null;
        int value = 0;

        while (l1 != null || l2 != null || value > 0) {
            value += (l1 != null ? l1.val : 0) + (l2 != null ? l2.val : 0);

            var newNode = new ListNode(value % 10, null);
            if (prev != null) {
                prev.next = newNode;
            } else {
                head = newNode;
            }
            prev = newNode;

            l1 = (l1 != null ? l1.next : null);
            l2 = (l2 != null ? l2.next : null);
            value /= 10;
        }
        return head;
    }
}
